src/gpu/intel/sycl/l0/utils.cpp

```diff
@@ -272,6 +287,113 @@ bool compare_ze_devices(const ::sycl::device &lhs, const ::sycl::device &rhs) {
     return lhs_ze_handle == rhs_ze_handle;
 }
 
+status_t get_device_ip(ze_device_handle_t device, uint32_t &ip_version) {
+    ze_device_properties_t deviceProps = {ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES};
+    ze_device_ip_version_ext_t devicePropsIP
+            = {ZE_STRUCTURE_TYPE_DEVICE_IP_VERSION_EXT};
+    deviceProps.pNext = &devicePropsIP;
+    CHECK(func_zeDeviceGetProperties(device, &deviceProps));
+    ip_version = devicePropsIP.ipVersion;
+    return status::success;
+}
+
+status_t get_l0_device_enabled_systolic_intel(
+        ze_device_handle_t device, bool &mayiuse_systolic) {
+    ze_device_module_properties_t deviceModProps
+            = {ZE_STRUCTURE_TYPE_DEVICE_MODULE_PROPERTIES};
+    // Note: supported by Intel Driver 24.05 and onwards
+    ze_intel_device_module_dp_exp_properties_t deviceModPropsExt
+            = {ZE_STRUCTURE_INTEL_DEVICE_MODULE_DP_EXP_PROPERTIES};
+    deviceModProps.pNext = &deviceModPropsExt;
+
+    CHECK(func_zeDeviceGetModuleProperties(device, &deviceModProps));
```

Comment:
Do we have a driver support matrix? Users may have pretty old drivers (esp. on Linux), so I'm wondering if we need a fallback path here.

Reply:
For backward compatibility, we validate only last driver available at time of release (see [README](https://github.com/oneapi-src/oneDNN/blob/main/README.md#validated-configurations)). In this particular case, this query appeared on Oct 31st 2023 ([99abb40a](https://github.com/intel/compute-runtime/commit/99abb40a45f51c8ea9779179e771c04a8be2777a))

For forward compatibility, I am working on getting clarification (and this PR will remain draft in the meantime).

Reply:
So I clarified the backward compatibility policy with Level Zero driver team. The extensions are backward compatible for a given major version of level zero API. So this is similar to regular (non-extensions) level zero APIs we currently use, and we should be fine with future driver updates.

However, this extension was introduced with [23.48.27912.11](https://github.com/intel/compute-runtime/releases/tag/23.48.27912.11) release, so this would become a requirement to our users. Do you think this could be an issue for some users?

Reply:
> However, this extension was introduced with [23.48.27912.11](https://github.com/intel/compute-runtime/releases/tag/23.48.27912.11) release, so this would become a requirement to our users. Do you think this could be an issue for some users?

I don't have any hard data on the range of L0 driver versions users have in the wild, but it seems risky to drop support for drivers more than 1 year old, especially as we have been gracefully supporting much older drivers prior to this PR. At least, let's not fail engine/primitive initialization if this call fails. We could conservatively report "no systolic support" for these older drivers. What do you think?

Reply:
+ @vpirogov. Do we need/have a policy wrt to number of drivers oneDNN supports?

Reply:
oneAPI policy is supporting the latest LTS and the latest rolling driver releases. I don't have concerns with dropping support for drivers >1 year old. Though I agree with @petercad's comment that we should fail gracefully if the driver is not supported. We could either assume that query returned negative or throw an error indicating that driver is not supported.